formula/src/main/java/com/instacart/formula/FormulaContext.kt

```diff
@@ -21,7 +21,7 @@ abstract class FormulaContext<State> internal constructor(
      * It uses [transition] type as key.
      */
     fun <Event> onEvent(
-        transition: Transition.Factory.(Event) -> Transition<State>,
+        transition: Transition<State, Event>,
```

Comment:
looks way more understandable than before 👍 

Transition.Factory.(Unit) -> Transition<State>
does not scan as something very intuitive